yeastphenome/apps/datasets/views.py

```diff
@@ -268,48 +251,34 @@ def gene_detail(request, query):
 def similar_genes(request, gene_id):
 
     try:
-        gene = Gene.objects.get(id=gene_id)
+        gene = Gene.objects.get(pk=gene_id)
     except Gene.DoesNotExist:
         raise Http404
 
-    sims = (
-        GeneSimilarity.objects.filter(gene1=gene)
-        .values_list(
-            "gene2__systematic_name",
-            "gene2__common_name",
-            "score",
-            "pvalue",
-            "gene2__id",
-        )
-        .order_by("-score")
-        .distinct()
-    )
+    sims = gene.get_ranked_similar(reverse=False).select_related("gene2")
 
     links = [
         {"url": reverse("common:explorer"), "name": "Explore data"},
         {"url": reverse("genes:index"), "name": "Genes"},
         {
-            "url": reverse("genes:detail", args=[gene.systematic_name]),
-            "name": "%s/%s" % (gene.common_name, gene.systematic_name),
+            "url": reverse("genes:detail", args=[gene.id]),
+            "name": "%s" % gene,
         },
         {
-            "url": reverse("genes:similar_genes", args=[gene.systematic_name]),
+            "url": reverse("genes:similar_genes", args=[gene.id]),
             "name": "Similar Genes",
         },
     ]
 
     total_sims = sims.count()
     ranks = [(1 - (idx / total_sims)) * 100 for idx, sim in enumerate(sims)]
-    context = get_gene_names_context()
-    context.update(
-        {
-            "gene": gene,
-            "sims": sims,
-            "ranks": ranks,
-            "links": links,
-            "active": "explorer",
-        }
-    )
+    context = {
```

Comment:
Can you explain why you are removing the original context that has all of the genes / aliases? Does the view here still work?

Reply:
Yeah, it all works . I removed it because, as far as I can tell, you need that context in this view. You have pretty much everything you need in the `sims` queryset. Maybe the flipped question is: is there anything I'm missing? why do we need a list of genes/aliases here (at the cost of running an extra query on the db)?

Reply:
This is correct! I was thinking of a different view. I'm not sure why I added it here, perhaps preparing to add a chart to the similar genes view. I can add this later if needed.